### PR TITLE
fix: runner updates were broken by a last minute change

### PR DESCRIPTION
**Pull request by alecthomas**
There were essentially two code paths that were used for runner->controlplane updates: the initial request, and subsequent updates. I've unified this into a single UpsertRunner() method on the DAL.

Also fix a bug where route selection failed if there were multiple deployments

### CODE REVIEW

File: controlplane/controlplane.go [hunk 1]
```diff
@@ -157,15 +155,18 @@ func (s *Service) RegisterRunner(ctx context.Context, req *connect.ClientStream[
...
+		return nil, connect.NewError(connect.CodeInvalidArgument, errors.WithStack(err))
+	}
+	err = s.dal.UpsertRunner(ctx, dal.Runner{
+		Key:        runnerKey,
+		Language:   msg.Language,
+		Endpoint:   msg.Endpoint,
+		State:      dal.RunnerStateFromProto(msg.State),
+		Deployment: maybeDeployment,
+	}) //
+	// runnerKey, msg.Language, endpoint)
```
Comment (wesbillman) [inline controlplane/controlplane.go:169]: Are these left over comments?

File: controlplane/controlplane.go [hunk 2]
```diff
@@ -206,26 +207,29 @@ func (s *Service) RegisterRunner(ctx context.Context, req *connect.ClientStream[
...
 			}
-			err = s.dal.UpdateRunner(ctx, runnerKey, toState, deploymentKey)
+			err = s.dal.UpsertRunner(ctx, dal.Runner{
+				Key:        runnerKey,
+				Language:   msg.Language,
+				Endpoint:   msg.Endpoint,
+				State:      toState,
+				Deployment: maybeDeployment,
+			})
+			// runnerKey, toState, deploymentKey)
```
Comment (wesbillman) [inline controlplane/controlplane.go:232]: Same question here.

File: controlplane/internal/dal/dal.go [hunk 3]
```diff
@@ -215,25 +214,33 @@ func (d *DAL) GetLatestDeployment(ctx context.Context, module string) (*Deployme
...
+		pgDeploymentKey.Bytes = dkey
+	}
+	deploymentID, err := d.db.UpsertRunner(ctx, sql.UpsertRunnerParams{
+		Key:           sqltypes.Key(runner.Key),
+		Language:      runner.Language,
+		Endpoint:      runner.Endpoint,
+		State:         sql.RunnerState(runner.State),
+		DeploymentKey: pgDeploymentKey,
+	})
+	// sqltypes.Key(key), language, endpoint.String())
```
Comment (wesbillman) [inline controlplane/internal/dal/dal.go:234]: And here. Maybe these are intentional?

**Review by wesbillman — APPROVED** [review]

File: controlplane/internal/dal/dal.go (hunk 3 above)
Comment (alecthomas) [reply controlplane/internal/dal/dal.go:234]: Aha whoops! These were leftover during refactoring!

**Pull request merged**